tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -1051,44 +1048,59 @@ class DxcCompiler : public IDxcCompiler3,
             pSourceInfo = debugSourceInfoWriter.GetPart();
           }
 
-          CComPtr<AbstractMemoryStream> pReflectionStream;
-          uint32_t reflectionSizeInBytes = 0;
-
           CComPtr<IDxcBlob> pDebugProgramBlob;
+          CComPtr<AbstractMemoryStream> pReflectionInPdb;
           // Don't include the debug part if using source only PDB
           if (opts.SourceOnlyDebug) {
             assert(pSourceInfo);
-            hlsl::ReEmitLatestReflectionData(compiledModule.get());
-            hlsl::StripAndCreateReflectionStream(compiledModule.get(), &reflectionSizeInBytes, &pReflectionStream);
+            pReflectionInPdb = pReflectionStream;
           }
           else {
             if (!opts.SourceInDebugModule) {
               // Strip out the source related metadata
-              compiledModule->GetOrCreateDxilModule()
+              debugModule->GetOrCreateDxilModule()
```

Comment:
Could `debugModule` be null if `opts.SourceOnlyDebug()` is true?  I think this would crash.

Reply:
Oh, never mind we are under the else for if (opts.SourceOnlyDebug).